lib/imagemounter/utils/list_developer_image.js

```diff
@@ -0,0 +1,184 @@
+ * @property {string} developerImageSignature
+ */
+
+/**
+ * Find developer image for certain version. If developer image does not exists,
+ * this will try to find and download developer image, unzip to `~/.appium/iOSSupport/`
+ * @param {string} version full version of iOS device. The first two parts of version
+ * will be preserved when sending the request, e.g. `14.7.1` will be changed to `14.7`
+ * @returns {ImagePath}
+ * @throws If developer image is not found, or error while downloading or unzipping.
+ */
+async function findDeveloperImage(version) {
+    const finalVersion = version.split('.').splice(0, 2).join('.');
+    const fileName = `${finalVersion}.zip`;
+    const DEFAULT_IMAGE_DIR = joinPath(await env.resolveAppiumHome(), DEFAULT_IMAGE_DIR_NAME);
+    const fullDownloadPath = joinPath(DEFAULT_IMAGE_DIR, fileName);
+    if (!await exists(DEFAULT_IMAGE_DIR)) {
+        await mkdir(DEFAULT_IMAGE_DIR, { recursive: true });
+    }
+    if (!await exists(fullDownloadPath)) {
+        await searchAndDownloadDeveloperImageFromGithub(finalVersion, fullDownloadPath);
+    }
+    const decompressPath = joinPath(DEFAULT_IMAGE_DIR, finalVersion);
+    let developerImageParentFolder = null;
+    if (await exists(decompressPath)) {
+        developerImageParentFolder = await findDeveloperImageFromDirectory(decompressPath);
+    }
+    if (!developerImageParentFolder) {
+        await zip.extractAllTo(fullDownloadPath, decompressPath);
+        developerImageParentFolder = await findDeveloperImageFromDirectory(decompressPath);
+    }
+    if (!developerImageParentFolder) {
+        throw new Error(`Unable to find unzipped developer image in ${decompressPath}`);
+    }
+    return {
+        developerImage: joinPath(developerImageParentFolder, DEVELOPER_IMAGE_FILE_NAME),
+        developerImageSignature: joinPath(developerImageParentFolder, DEVELOPER_IMAGE_SIGNATURE_FILE_NAME)
+    };
+}
+
+async function searchAndDownloadDeveloperImageFromGithub(finalVersion, fullDownloadPath) {
+    const fileNameRegExp = new RegExp(`${finalVersion}(\\(([\\w_|.()])+\\))?.zip`);
+    let fileUrl = null;
+    for (const repo of AVAILABLE_REPOS) {
+        const fileList = await listGithubImageList(repo.user, repo.name, repo.subFolderList);
+        if (!fileList) {
+            continue;
+        }
+        const targetFile = _.find(fileList, (item) => fileNameRegExp.test(item.path));
+        if (targetFile) {
+            fileUrl = `https://raw.githubusercontent.com/${repo.user}/${repo.name}/master/${repo.subFolderList.join('/')}/${targetFile.path}`;
+            break;
+        }
+    }
+    if (fileUrl === null) {
+        throw new Error(`Failed to get developer image for iOS ${finalVersion}`);
+    }
+    try {
+        log.info(`Downloading developer image for ${finalVersion} to ${fullDownloadPath} from ${fileUrl}`);
+        await net.downloadFile(fileUrl, fullDownloadPath);
+    } catch (e) {
+        await unlink(fullDownloadPath);
```

Comment:
not sure if this function throws an exception if the file does not exist. It would be safer to use fs.rimraf instead

Reply:
I see some todo annotation on `@appium/support/fs` which indicates  Node.js v12 support is dropped.
Will that be OK to use `rimraf` from here?

Reply:
yes. We do not support EOLed node versions. Version 12 has been EOLed long time ago. Version 14 will be EOLed at the end of April

Reply:
use [rm](https://nodejs.org/api/fs.html#fsrmpath-options-callback) instead.  assuming `@appium/support`'s `fs` has it.